### ISSUE TITLE
Json is not abstract and does not override abstract methods

### ISSUE THREAD

**Issue by rtedin** [immutables/immutables#217]
The following example does not compile:

```java
public interface Foo {
    String getId();
}

@Value.Immutable
@Value.Style (
        implementationNestedInBuilder = true,
        visibility = Value.Style.ImplementationVisibility.PACKAGE,
        builderVisibility = Value.Style.BuilderVisibility.PUBLIC
)
@JsonDeserialize // Note this Json annotation!
abstract class AbstractFoo implements Foo {
}
```

The compiler error is:

```
Json is not abstract and does not override abstract method getId() in Foo
```

This is just a minimal non-working example. I'm using this pattern in my code with meaningful values for @JsonDeserialize. I'm using the 2.1.1 release.

**Comment by elucash** [comment]
got it, this case is not covered, will fix it shortly. Thank you for the report!

**Comment by elucash** [comment]
2.1.2 is available, could you, please, check the fix?

**Comment by rtedin** [comment]
Meanwhile I have changed my code and I do not use `@JsonDeserialize` directly. But I've tested the example above and others and it seems to work. Thanks!

**Comment by elucash** [comment]
fixed in 2.1.3
